src/variant_detection/variant_detection.cpp

```diff
@@ -72,9 +83,18 @@ void detect_junctions_in_short_reads_sam_file(std::vector<Junction> & junctions,
             seqan3::debug_stream << num_good << " good alignments from short read file." << std::endl;
         }
     }
+
+    size_t i = 0;
+    for(std::string ref_id : ref_ids)
+    {
+        sn_ln_tag_tuple_set.insert({ref_id, std::get<0>(ref_id_info[i])});
```

Comment:
Avoid copying of std::string by using `std::string const&`.
Using emplace which will create the tuple in-place.
```suggestion
    for(std::string const& ref_id : ref_ids)
    {
        sn_ln_tag_tuple_set.emplace(ref_id, std::get<0>(ref_id_info[i]));
```